pkg/solana/logpoller/orm.go

```diff
@@ -72,13 +74,42 @@ func (o *DSORM) InsertFilter(ctx context.Context, filter Filter) (id int64, err
 
 // GetFilterByID returns filter by ID
 func (o *DSORM) GetFilterByID(ctx context.Context, id int64) (Filter, error) {
-	query := `SELECT id, name, address, event_name, event_sig, starting_block, event_idl, subkey_paths, retention, max_logs_kept
-		FROM solana.log_poller_filters WHERE id = $1`
+	query := filtersQuery("WHERE id = $1")
 	var result Filter
 	err := o.ds.GetContext(ctx, &result, query, id)
 	return result, err
 }
 
+func (o *DSORM) MarkFilterDeleted(ctx context.Context, id int64) (err error) {
+	query := `UPDATE solana.log_poller_filters SET is_deleted = true WHERE id = $1`
+	_, err = o.ds.ExecContext(ctx, query, id)
+	return err
+}
+
+func (o *DSORM) DeleteFilter(ctx context.Context, id int64) (err error) {
+	query := `DELETE FROM solana.log_poller_filters WHERE id = $1`
+	_, err = o.ds.ExecContext(ctx, query, id)
+	return err
+}
+
+func (o *DSORM) DeleteFilters(ctx context.Context, filters []Filter) error {
+	for _, filter := range filters {
+		err := o.DeleteFilter(ctx, filter.ID)
```

Comment:
We might we need to create a transaction for this, and pass it down to `DeleteFilter`.
What happens if some filters are deleted properly but then partway through this for loop the network connection to the db momentarily goes out and returns an error?  It would return from `DeleteFilters` with an error at that point, with some deleted and some still there.

Reply:
(Alternatively, could do a single DELETE query with a list of all the filter ID's to delete.)

Reply:
Even if db fails when only part of the filters is deleted, it's safe for us to retry as we are using ID. We'll add all filters scheduled for removal back into the queue and try again.
I hesitate to wrap it into the transaction as in case of timeout due to a large number of filters to be removed; we'll get stuck trying to delete filters and rolling back due to time.

Reply:
Alright, I guess that's good enough.